config/samples/dataplane_v1beta1_openstackdataplane_nmstate.yaml

```diff
@@ -0,0 +1,258 @@
+spec:
+  deployStrategy:
+    deploy: false
+  nodes:
+    edpm-compute-0:
+      role: edpm-compute
+      hostName: edpm-compute-0
+      ansibleHost: 192.168.122.100
+      node:
+        ansibleVars: |
+          ctlplane_ip: 192.168.122.100
+          internal_api_ip: 172.17.0.100
+          storage_ip: 172.18.0.100
+          tenant_ip: 172.19.0.100
+          fqdn_internal_api: edpm-compute-0.example.com
+          # edpm_network_config - nmstate
+          edpm_network_config_tool: 'nmstate'
+          network_state:
+            dns-resolver:
+              config:
+                server:
+                - 192.168.122.80
+            interfaces:
+              - name: eth0
+                type: ethernet
+                state: up
+              - name: br-ex
+                type: ovs-interface
+                state: up
+                mtu: 1500
+                ipv4:
+                  enabled: true
+                  address:
+                    - ip: "192.168.122.100"
+                      prefix-length: 24
+              - name: vlan20
+                type: ovs-interface
+                state: up
+                mtu: 1500
+                ipv4:
+                  enabled: true
+                  address:
+                    - ip: "172.17.0.100"
+                      prefix-length: 24
+              - name: vlan21
+                type: ovs-interface
+                state: up
+                mtu: 1500
+                ipv4:
+                  enabled: true
+                  address:
+                    - ip: "172.18.0.100"
+                      prefix-length: 24
+              - name: vlan22
+                type: ovs-interface
+                state: up
+                mtu: 1500
+                ipv4:
+                  enabled: true
+                  address:
+                    - ip: "172.19.0.100"
+                      prefix-length: 24
+              - name: br-ex
+                type: ovs-bridge
+                bridge:
+                  options:
+                    fail-mode: standalone
+                  port:
+                  - name: eth0
+                  - name: br-ex
+                  - name: vlan20
+                    vlan:
+                      mode: access
+                      tag: 20
+                  - name: vlan21
+                    vlan:
+                      mode: access
+                      tag: 21
+                  - name: vlan22
+                    vlan:
+                      mode: access
+                      tag: 22
+            routes:
+              config:
+              - destination: 0.0.0.0/0
+                next-hop-address: 192.168.122.1
+                next-hop-interface: br-ex
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+      deployStrategy:
+        deploy: false
+    edpm-compute-1:
+      role: edpm-compute
+      hostName: edpm-compute-1
+      ansibleHost: 192.168.122.101
+      node:
+        ansibleVars: |
```

Comment:
Do we need to add `edpm_network_config_nmstate: true` to the vars?
https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_network_config/tasks/network_config.yml#L154

And yeah, just reiterating, but I think this would be cleaner if nested under `networkConfig`